### PR TITLE
Feature/eng 1692 support quadratic obj functions for mips

### PR DESCRIPTION
**Pull request by dirkschumacher**
This adds support for quadratic terms in the objective function

### CODE REVIEW

File: mip/objective.go [hunk 1]
```diff
@@ -10,12 +10,15 @@ package mip
 //
 // 2.5 * x and 3.5 * y are 2 terms in this example.
 type Objective interface {
-	// IsMaximize return true if the invoking objective is a maximization
+	IsLinear() bool
```
Comment (larsbeck) [inline mip/objective.go]: Is this lacking a comment?

**Review by larsbeck — DISMISSED** [review]

File: mip/objective.go (hunk 1 above)
Comment (dirkschumacher) [reply mip/objective.go]: 👍

File: mip/objective.go [hunk 2]
```diff
@@ -24,6 +28,23 @@ type Objective interface {
...
+	//
+	//      m.Objective().SetMaximize()
+	//      // results in: maximize -
+	//      m.Objective().NewQuadraticTerm(1.0, x1, x1)
+	//      // results in: maximize 1.0 * x1^2
+	//      m.Objective().NewQuadraticTerm(1.0, x1, x2)
+	//      // results in: maximize 1.0 * x1^2 + x1x2
+	//      m.Objective().NewQuadraticTerm(1.0, x2, x1)
+	//      // results in: maximize 1.0 * x1^2 + 2.0 * x1x2
+	NewQuadraticTerm(coefficient float64, variable1, variabl2 Var) QuadraticTerm
```
Comment (davidrijsman) [inline mip/objective.go]: Typo variabl2

File: mip/term.go [hunk 3]
```diff
@@ -7,9 +7,24 @@ package mip
...
+	// Var returns the variable of the term.
 	Var() Var
 }
 
 // Terms is a slice of Term instances.
 type Terms []Term
+
+// QuadraticTerm consists of a coefficient and two vars. It should be
+// interpreted as the product of a coefficient and the two vars in the context
+// of the objective.
```
Comment (davidrijsman) [inline mip/term.go]: in the context ...is not required I think

**Review by davidrijsman — DISMISSED** [review]

**Review by sebastian-quintero — DISMISSED** [review]
What David said.

**Review by sebastian-quintero — DISMISSED** [review]

**Review by davidrijsman — APPROVED** [review]

**Review by sebastian-quintero — APPROVED** [review]

**Review by merschformann — APPROVED** [review]
Great to have quadratic support!

square-bear

**Pull request merged**